drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -875,23 +875,22 @@ static int lsm6dsl_init(const struct device *dev)
 	})
 
 #ifdef CONFIG_LSM6DSL_TRIGGER
-#define LSM6DSL_CONFIG_SPI(inst)					\
-	{								\
-		.bus_name = DT_INST_BUS_LABEL(inst),			\
-		.bus_init = lsm6dsl_spi_init,				\
-		.bus_cfg = { .spi_cfg = LSM6DSL_SPI_CFG(inst)	},	\
+#define LSM6DSL_CFG_IRQ(inst) \
 		.irq_dev_name = DT_INST_GPIO_LABEL(inst, irq_gpios),	\
 		.irq_pin = DT_INST_GPIO_PIN(inst, irq_gpios),		\
-		.irq_flags = DT_INST_GPIO_FLAGS(inst, irq_gpios),	\
-	}
+		.irq_flags = DT_INST_GPIO_FLAGS(inst, irq_gpios),
 #else
+#define LSM6DSL_CFG_IRQ(inst)
+#endif /* CONFIG_LSM6DSL_TRIGGER */
+
 #define LSM6DSL_CONFIG_SPI(inst)					\
 	{								\
 		.bus_name = DT_INST_BUS_LABEL(inst),			\
 		.bus_init = lsm6dsl_spi_init,				\
-		.bus_cfg = { .spi_cfg = LSM6DSL_SPI_CFG(inst)	}	\
+		.bus_cfg = { .spi_cfg = LSM6DSL_SPI_CFG(inst)	},	\
+		COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, irq_gpios),	\
```

Comment:
Some people are using:

    IF_ENABLED(DT_INST_NODE_HAS_PROP(inst, irq_gpios),	\
               (LSM6DSL_CFG_IRQ(inst)))

for this sort of thing on the view that it's less opaque than `COND_CODE_1`.  We don't have an expectation of that AFAIK.